packages/app/src/lib/import.ts

```diff
@@ -536,6 +570,229 @@ export async function isBitwarden(file: File): Promise<boolean> {
+}
+
+export async function asDashlane(file: File): Promise<VaultItem[]> {
+    const data = await readFileAsText(file);
+    const papa = await loadPapa();
+    const items = papa
+        .parse(data)
+        .data // Remove first row as it only contains field names
+        .slice(1)
+        // Filter out empty rows
+        .filter((row: string[]) => row.length > 1)
+        .map(dashlaneParseRow);
+
+    return Promise.all(items);
+}
+
+export async function isDashlane(file: File): Promise<boolean> {
+    try {
+        const data = await readFileAsText(file);
+        const headerRow = data.split("\n")[0].trim();
+        return headerRow === "username,username2,username3,title,password,note,url,category,otpSecret";
+    } catch (error) {
+        return false;
+    }
+}
+
+async function keePassParseRow(row: string[]): Promise<VaultItem> {
+    const nameIndex = 0;
+    const urlIndex = 3;
+    const usernameIndex = 1;
+    const passwordIndex = 2;
+    const notesIndex = 4;
+
+    const fields: Field[] = [
+        new Field({ name: $l("Username"), value: row[usernameIndex], type: FieldType.Username }),
+        new Field({ name: $l("Password"), value: row[passwordIndex], type: FieldType.Password }),
+        new Field({ name: $l("URL"), value: row[urlIndex], type: FieldType.Url }),
+    ];
+
+    if (row[notesIndex]) {
+        fields.push(new Field({ name: $l("Notes"), value: row[notesIndex], type: FieldType.Note }));
+    }
+
+    return createVaultItem({ name: row[nameIndex], fields, tags: [] });
+}
+
+export async function asKeePass(file: File): Promise<VaultItem[]> {
+    const data = await readFileAsText(file);
+    const papa = await loadPapa();
+    const items = papa
+        .parse(data)
+        .data // Remove first row as it only contains field names
+        .slice(1)
+        // Filter out empty rows
+        .filter((row: string[]) => row.length > 1)
+        .map(keePassParseRow);
+
+    return Promise.all(items);
+}
+
+export async function isKeePass(file: File): Promise<boolean> {
+    try {
+        const data = await readFileAsText(file);
+        const headerRow = data.split("\n")[0].trim();
+        return headerRow === '"Account","Login Name","Password","Web Site","Comments"';
+    } catch (error) {
+        return false;
+    }
+}
+
+async function nordPassParseRow(row: string[]): Promise<VaultItem> {
+    const nameIndex = 0;
+    const urlIndex = 1;
+    const usernameIndex = 2;
+    const passwordIndex = 3;
+    const notesIndex = 4;
+    const cardHolderNameIndex = 5;
+    const cardNumberIndex = 6;
+    const cvcIndex = 7;
+    const expiryDateIndex = 8;
+    const zipCodeIndex = 9;
+    const folderIndex = 10;
+    const fullNameIndex = 11;
+    const phoneIndex = 12;
+    const emailIndex = 13;
+    const address1Index = 14;
+    const address2Index = 15;
+    const cityIndex = 16;
+    const countryIndex = 17;
+    const stateIndex = 18;
```

Comment:
This could've been a map with the index, field name, and field type, but it was only saving 10-20 lines and this is very readable, just long.